crypto/ex_data.c

```diff
@@ -287,7 +287,7 @@ int CRYPTO_dup_ex_data(int class_index, CRYPTO_EX_DATA *to,
         CRYPTOerr(CRYPTO_F_CRYPTO_DUP_EX_DATA, ERR_R_MALLOC_FAILURE);
         return 0;
     }
-    if (!CRYPTO_set_ex_data(to, mx - 1, NULL))
+    if (!CRYPTO_set_ex_data(to, mx - 1, CRYPTO_get_ex_data(to, mx - 1)))
```

Comment:
Hrm? Maybe I did not divine the reason for this line correctly (more reason the original commit should have had a comment!). My current assumption is that this semantically no-op `CRYPTO_set_ex_data`'s purpose is to ensure that the unchecked `CRYPTO_set_ex_data` below cannot fail because, were it to fail, `ptr` would be leaked with no way to free it.